anesthetic/plotting/_matplotlib/core.py

```diff
@@ -0,0 +1,45 @@
+from pandas.plotting._matplotlib.core import (ScatterPlot as _ScatterPlot,
+                                              HexBinPlot as _HexBinPlot,
+                                              MPLPlot)
+from anesthetic.weighted_pandas import _WeightedObject
+
+
+def _get_weights(kwargs, data):
+    if isinstance(data, _WeightedObject):
+        kwargs['weights'] = data.weights
+
+
+class _WeightedMPLPlot(MPLPlot):
+    def __init__(self, data, *args, **kwargs):
+        _get_weights(kwargs, data)
+        super().__init__(data, *args, **kwargs)
+
+
+def _compress_weights(kwargs, data):
+    if isinstance(data, _WeightedObject):
+        return data.compress(kwargs.pop('ncompress', None))
+    else:
+        return data
+
+
+class _CompressedMPLPlot(MPLPlot):
+    def __init__(self, data, *args, **kwargs):
+        data = _compress_weights(kwargs, data)
+        super().__init__(data, *args, **kwargs)
+
+
+class ScatterPlot(_CompressedMPLPlot, _ScatterPlot):
+    # noqa: disable=D101
+    def __init__(self, data, x, y, s=None, c=None, **kwargs) -> None:
+        if isinstance(data, _WeightedObject):
+            kwargs['alpha'] = kwargs.get('alpha', 0.5)
+        super().__init__(data, x, y, s, c, **kwargs)
+
+
+class HexBinPlot(_HexBinPlot):
+    # noqa: disable=D101
+    def __init__(self, data, x, y, C=None, **kwargs) -> None:
+        if isinstance(data, _WeightedObject):
+            C = '__weights'
+            data[C] = data.weights
```

Comment:
Compute a weighted hexbin plot by using the C functionality to accumulate weight rather than counts in each hexbin.

Reply:
Looks like `C` is what will be used with a colourbar. What type of normalisation would we want by default for that?

Reply:
Why do `HistPlot` and `KDEPlot` get their own python file, but `ScatterPlot` and `HexBinPlot` are put in `core.py`?

Reply:
This is consistent with the `pandas.plotting._matplotlib` structure, see e.g. [ScatterPlot](https://github.com/pandas-dev/pandas/blob/v1.4.3/pandas/plotting/_matplotlib/core.py#L1055-#L1140).

Reply:
> Looks like `C` is what will be used with a colourbar. What type of normalisation would we want by default for that?

How do we prefer to treat the normalisation for colourbars? 

For the 1-dim case we have a kwarg `density`, with `density=False` normalising such that the min is 0 and the max is 1, and with `density=True` normalising such that the integral is 1.

Reply:
Ah, I see. At the moment I think we probably want `WeightedDataFrame` to be consistent with what `DataFrame` does, namely to normalise the colour to the maximum weight. If we wanted to implement a `density` keyword, we should implement our own `anesthetic.plot.hexbin_plot_2d` wrapper as we do for KDE and hists.